src/callbacks/onPreprocess/flattenData/calculateRatios.js

```diff
@@ -0,0 +1,25 @@
+export default function calculateRatios(d, participant_obj) {
+    this.config.measure_details.forEach(d => {
+        this.config.measure_details.filter(di => di.measure !== d.measure).forEach(di => {
+            participant_obj[`${d.label}_relative_uln/${di.label}_relative_uln`] =
+                participant_obj[`${d.label}_relative_uln`] /
+                participant_obj[`${di.label}_relative_uln`];
+        });
+    });
+
+    //R-ratio should be the ratio of ALT to ALP, i.e. the x-axis to the z-axis.
```

Comment:
Not sure the z-axis concept is in use in any other parts of the chart at this point, so will probably re-factor this to just use the label values "ALT" and "ALP" instead.